launcher/resources/git-shim.py

```diff
@@ -59,10 +48,19 @@ def handle_intrinsic(root):
 cmd = json.dumps({ "workingDir": relative_dir, "cmd": ["git"] + sys.argv[1:] })
 s.send(cmd.encode('utf-8') + '\n')
 
-reply = receive_all(s).decode("utf-8")
-s.close()
+socket_file = s.makefile()
+
+for line in socket_file.readlines():
+    response = json.loads(line)
+    if len(response) == 2:
+        if response[0] == 0:
+            print(response[1])
```

Comment:
What's the newline behavior in this case, should this be `print(.., end="")`?

Reply:
It seems that the `line` being read from `socket_file` doesn't have the trailing `\n`, so if I left the `end=""` in it was squishing everything into a single line

Reply:
Makes sense